docs/data/material/customization/css-layers/css-layers.md

```diff
@@ -0,0 +1,250 @@
+Then follow these steps:
+
+1. Enable the [CSS layer feature](/material-ui/integrations/nextjs/#configuration-2) in a custom `_document`:
+
+```tsx title="pages/_document.tsx"
+import {
+  createCache,
+  documentGetInitialProps,
+} from '@mui/material-nextjs/v15-pagesRouter';
+
+// ...
+
+MyDocument.getInitialProps = async (ctx: DocumentContext) => {
+  const finalProps = await documentGetInitialProps(ctx, {
+    emotionCache: createCache({ enableCssLayer: true }),
+  });
+  return finalProps;
+};
+```
+
+2. Configure the layer order with the `GlobalStyles` component to work with Tailwind CSS v4—it must be the first child of the `AppCacheProvider`:
+
+```tsx title="pages/_app.tsx"
+import { AppCacheProvider } from '@mui/material-nextjs/v15-pagesRouter';
+import GlobalStyles from '@mui/material/GlobalStyles';
+
+export default function MyApp(props: AppProps) {
+  const { Component, pageProps } = props;
+  return (
+    <AppCacheProvider {...props}>
+      <GlobalStyles styles="@layer theme, base, mui, components, utilities;" />
+      <Component {...pageProps} />
+    </AppCacheProvider>
+  );
+}
+```
+
+### Vite or any other SPA
+
+Make the following changes in `src/main.tsx`:
+
+1. Pass the `enableCssLayer` prop to the `StyledEngineProvider` component.
+2. Configure the layer order with the `GlobalStyles` component to work with Tailwind CSS v4.
+
+```tsx title="main.tsx"
+import { StyledEngineProvider } from '@mui/material/styles';
+import GlobalStyles from '@mui/material/GlobalStyles';
+
+ReactDOM.createRoot(document.getElementById('root')!).render(
+  <React.StrictMode>
+    <StyledEngineProvider enableCssLayer>
+      <GlobalStyles styles="@layer theme, base, mui, components, utilities;" />
+      {/* Your app */}
+    </StyledEngineProvider>
+  </React.StrictMode>,
+);
+```
+
+## Multiple layers
+
+On top of the [single layer](#single-layer), you can split styles into multiple layers to better organize them within Material UI.
+This makes theming and overriding styles through the `sx` prop easier.
+
+Follow the steps from the [previous section](#single-layer) to enable the CSS layer feature.
+Then, create a new file and export the component that wraps the `ThemeProvider` from Material UI.
+Finally, pass the `experimental_modularCssLayers: true` option to the `createTheme` function:
+
+```tsx title="src/theme.tsx"
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+
+const theme = createTheme({
+  experimental_modularCssLayers: true,
+});
+
+export default function AppTheme({ children }: { children: ReactNode }) {
+  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
+}
+```
+
+{{"demo": "CssLayersInput.js"}}
+
+When this feature is enabled, Material UI generates these layers:
+
+- `@layer mui.global`: The global styles from `GlobalStyles` and `CssBaseline` components.
+- `@layer mui.default`: The base styles for all Material UI components.
```

Comment:
Can we rename this to components? It could be more intuitive for people.

Reply:
+1, Tailwind also uses "components"